libfuse/file.go

```diff
@@ -73,6 +73,28 @@ func (f *File) fillAttrWithMode(
 	return nil
 }
 
+// Access implements the fs.NodeAccesser interface for File. This is necessary
+// for macOS to correctly identify plaintext file as plaintext. If not
+// implemented, bazil-fuse returns a nil error for every thing, so when macOS
+// checks for executable bit using Access (instead of Attr!), it gets a
+// success, which makes it think the file is executable, yielding a "Unix
+// executable" UTI.
+func (f *File) Access(ctx context.Context, r *fuse.AccessRequest) error {
+	if r.Mask&0111 != 0 {
```

Comment:
And actually, shouldn't we be returning an error for any permissions that aren't on the user bit?  I.e., even if it's an executable file, we should return an error for a mask like `0001`, right?